File: Data_Structure_using_JavaScript/Stack.js
class node
{
    data
    next

    constructor(no)
    {
        this.data = no
        this.next = null
    }
}

class Stack
{
    #first
    #iCount

    constructor()
    {
        this.#first = null
        this.#iCount = 0
    }

    Push(iNo)           // InsertFirst
    {
        let newn = null

        newn = new node(iNo)

        newn.next = this.#first
        this.#first = newn

        this.#iCount++
    }
    
    Pop()               // DeleteFirst
    {
        let element = 0

        if(null == this.#first)
        {
            console.log("Stack is empty")
            return
        }
        else
        {
            element = this.#first.data
            this.#first = this.#first.next
            this.#iCount--
            return element
        }
    }

    Display()
    {
        let temp = null
        temp = this.#first

        while(temp != null)
        {
            console.log("| "+temp.data+" |")
            temp = temp.next
        }
    }

    Count()
    {
        return this.#iCount
    }
}

console.log()
console.log("---------------------------------- For Number ----------------------------------------")
console.log()

const sobj = new Stack()
let iRet = 0

sobj.Push(11)
sobj.Push(21)
sobj.Push(51)
sobj.Push(101)
sobj.Push(111)
sobj.Push(121)
sobj.Push(151)

sobj.Display()

iRet = sobj.Count()

console.log("Number of nodes in Stack is : "+iRet)

iRet = sobj.Pop()

console.log("Poped element is : "+iRet)

iRet = sobj.Pop()

console.log("Poped element is : "+iRet)

sobj.Display()

iRet = sobj.Count()

console.log("Number of nodes in Stack is : "+iRet)

/////////////////////////////////////////////////////////////////////////////

console.log()
console.log("---------------------------------- For Bigint ----------------------------------------")
console.log()

const sobj2 = new Stack()

sobj2.Push(11874685466666n)
sobj2.Push(21876496896854n)
sobj2.Push(51635465643165n)
sobj2.Push(10164361166635n)
sobj2.Push(11196683565266n)
sobj2.Push(12168163656565n)
sobj2.Push(15136354636565n)

sobj2.Display()

iRet = sobj2.Count()

console.log("Number of nodes in Stack is : "+iRet)

iRet = sobj2.Pop()

console.log("Poped element is : "+iRet)

iRet = sobj2.Pop()

console.log("Poped element is : "+iRet)

sobj2.Display()

iRet = sobj2.Count()

console.log("Number of nodes in Stack is : "+iRet)

File: Data_Structure_using_C_Programming/Stack.c
#include<stdio.h>
#include<stdlib.h>

struct node
{
    int data;
    struct node* next;
};

typedef struct node NODE;
typedef struct node* PNODE;
typedef struct node** PPNODE;

void Push(PPNODE first, int iNo)     // InsertFirst
{
    PNODE newn = NULL;

    newn = (PNODE)malloc(sizeof(NODE));
    newn->data = iNo;
    newn->next = NULL;

    newn->next = *first;
    *first = newn;
}

int Pop(PPNODE first)     // DeleteFirst
{
    int iValue = 0;
    PNODE temp = NULL;

    if(NULL == *first)
    {
        printf("LinkedList is empty");
        return 0;
    }
    else if((*first)->next == NULL)
    {
        iValue = (*first)->data;
        free(*first);
        *first = NULL;
    }
    else
    {
        temp = *first;
        iValue = (*first)->data;
        *first = temp->next;
        free(temp);
    }
    return iValue;
}

void Display(PNODE first)
{
    printf(" ----\n");
    while(first != NULL)
    {
        printf("| %d |\n",first->data);
        printf(" ----\n");
        first = first->next;
    }
}

int Count(PNODE first)
{
    int iCount = 0;

    while (first != NULL)
    {
        iCount++;
        first = first->next;
    }

    return iCount;
}

int main()
{
    PNODE head = NULL;
    int iRet = 0;

    Push(&head,11);
    Push(&head,21);
    Push(&head,51);
    Push(&head,101);

    Display(head);

    iRet = Count(head);
    printf("Total elements in Stack is : %d\n",iRet);
    
    iRet = Pop(&head);
    printf("Poped element is : %d\n",iRet);
    
    iRet = Pop(&head);
    printf("Poped element is : %d\n",iRet);

    Display(head);

    iRet = Count(head);
    printf("Total elements in Stack is : %d\n",iRet);

    return 0;
}